iothub/service/src/Jobs/ScheduledJobsClient.cs

```diff
@@ -0,0 +1,378 @@
+        }
+
+        internal ScheduledJobsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _credentialProvider = credentialProvider;
+            _hostName = hostName;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Explicitly open the ScheduledJobsClient instance.
+        /// </summary>
+        public virtual Task OpenAsync()
+        {
+            return TaskHelpers.CompletedTask;
+        }
+
+        /// <summary>
+        /// Closes the ScheduledJobsClient instance and disposes its resources.
+        /// </summary>
+        /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-jobs"/>.
+        public virtual Task CloseAsync()
+        {
+            return TaskHelpers.CompletedTask;
+        }
+
+        /// <summary>
+        /// Gets the job with the specified Id.
+        /// </summary>
+        /// <param name="jobId">Id of the job to retrieve</param>
+        /// <param name="cancellationToken">Task cancellation token</param>
+        /// <returns>The matching JobResponse object</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the provided job Id is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if the job Id is empty or whitespace.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-jobs"/>.
+        public virtual async Task<JobResponse> GetAsync(string jobId, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, jobId, nameof(GetAsync));
+
+            try
+            {
+                Argument.RequireNotNullOrEmpty(jobId, nameof(jobId));
+                cancellationToken.ThrowIfCancellationRequested();
+
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Get, GetJobUri(jobId), _credentialProvider);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
```

Comment:
Add `ConfigureAwait(false)`

Reply:
And elsewhere

Reply:
You still need to add ```.ConfigureAwait(false)``` after every async call that gets awaited @tmahmood-microsoft 

Here and everywhere else

```suggestion
                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
```